### PR TITLE
add docs comparing oc and kubectl

### PR DESCRIPTION
**Pull request by juanvallejo**
Adds documentation explaining the main differences between `oc` and `kubectl`.
Trello card: https://trello.com/c/BumbnGUY

cc @mfojtik @liggitt @derekwaynecarr

### CODE REVIEW

File: cli_reference/differences_oc_kubectl.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Existing users of `kubectl` can continue to use this binary with no changes to the api, but should consider upgrading
+to `oc` in order to gain the added functionality mentioned in the previous section.
+
+See xref:../cli_reference/get_started_cli.adoc#cli-reference-get-started-cli[Get Started with the CLI] for
+installation and setup instructions.
+
+== Obtaining Kubectl
+
+The `kubectl` binary is included alongside your download of the command line client-tools.
```
Comment (mfojtik) [inline cli_reference/differences_oc_kubectl.adoc]: it is not :-) all the recent tars we publish on github does no have symlink nor the copy of oc inside. I would state that copying it manually from oc binary is the primary form of delivery :-)

**Comment by juanvallejo** [comment]
cc @openshift/team-documentation

**Comment by vikram-redhat** [comment]
@juanvallejo thanks for this PR. Are we adding kubectl from 3.11 onwards or would this work in 3.10 docs as well.

**Comment by juanvallejo** [comment]
@vikram-redhat As it currently stands, copying `oc` and renaming it `kubectl` would also work for the 3.10 docs - but based on @derekwaynecarr 's [comment](https://github.com/openshift/openshift-docs/pull/10842#discussion_r203149546), it looks like consensus still has to be reached as to how `kubectl` will ultimately be obtained by users

**Comment by vikram-redhat** [comment]
@juanvallejo ok I will add a WIP on this PR then.

File: cli_reference/differences_oc_kubectl.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
+[[cli-reference-differences-oc-kubectl]]
+= Differences Between OC and Kubectl
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+
+A copy of `kubectl` is included with the {product-title} command line interface (CLI).
+Although there are several similarities between these two clients, this guide will
```
Comment (soltysh) [inline cli_reference/differences_oc_kubectl.adoc]: s/guide will aim/guide's aim is to

File: cli_reference/differences_oc_kubectl.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+aim to clarify the main reasons and scenarios for using one over the other.
+
+== Using OC
+
+The `oc` binary offers full support for all of the features offered by an {product-title} cluster.
```
Comment (soltysh) [inline cli_reference/differences_oc_kubectl.adoc]: Stronger message will be if we write something like: 

oc binary provides the exact same capabilities as kubectl binary, on top of that it's extended to natively support {product-title} features such as:

File: cli_reference/differences_oc_kubectl.adoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+It provides functionality not offered by `kubectl`, such as:
+
+- Full support for OpenShift resources (DeploymentConifgs, BuildConfigs, Routes, ImageStreams, ImageStreamTags, etc.).
+These are resources specific to {product-title}, and not native to Kubernetes.
```
Comment (soltysh) [inline cli_reference/differences_oc_kubectl.adoc]: s/not native/not available

File: cli_reference/differences_oc_kubectl.adoc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- Authentication: The `oc` binary offers a built-in `login` command which allows authentication
+into a cluster based on specified configuration. See
```
Comment (soltysh) [inline cli_reference/differences_oc_kubectl.adoc]: command which allows authentication ~~into a cluster based on specified configuration.~~ against a cluster.

File: cli_reference/differences_oc_kubectl.adoc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+xref:../dev_guide/authentication.adoc[developer authentication] and
+xref:../install_config/configuring_authentication.adoc[configuring authentication] for more information.
+- Additional commands, such as `new-app`, which make it easier to get new applications started using existing source code.
```
Comment (soltysh) [inline cli_reference/differences_oc_kubectl.adoc]: existing source code or pre-built images.

File: cli_reference/differences_oc_kubectl.adoc [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+== Using Kubectl
+
+The `kubectl` binary is provided as a means to support existing workflows and scripts for users coming from a DIY
+Kubernetes environment.
+
+Existing users of `kubectl` can continue to use this binary with no changes to the api, but should consider upgrading
+to `oc` in order to gain the added functionality mentioned in the previous section.
```
Comment (soltysh) [inline cli_reference/differences_oc_kubectl.adoc]: The move from `kubectl` to `oc` should be additionally simple that `oc` is built on top of `kubectl` which means that all that is required is to change the binary name from `kubectl` to `oc`.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: cli_reference/differences_oc_kubectl.adoc [hunk 8]
```diff
@@ -0,0 +1,40 @@
+[[cli-reference-differences-oc-kubectl]]
+= Differences Between OC and Kubectl
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+
+A copy of `kubectl` is included with the {product-title} command line interface (CLI).
```
[the remark this answers is not in the document]
Comment (juanvallejo) [reply cli_reference/differences_oc_kubectl.adoc]: Thanks, updated to `A copy of kubectl is included in {product-title} command line interface`

**Comment by juanvallejo** [comment]
@soltysh thanks, comments addressed. Additionally, I have removed the section that mentions "obtaining a kubectl binary by renaming the existing `oc` binary"

File: cli_reference/differences_oc_kubectl.adoc [hunk 9]
```diff
@@ -1,27 +1,27 @@
 [[cli-reference-differences-oc-kubectl]]
-= Differences Between OC and Kubectl
+= Why Use oc over kubectl?
```
Comment (soltysh) [inline cli_reference/differences_oc_kubectl.adoc]: Why use <- lowercase use

**Review by soltysh — APPROVED** [review]
One final nit and this lgtm.

**Comment by juanvallejo** [comment]
@soltysh thanks, review comment addressed

**Comment by soltysh** [comment]
@adellape ptal

**Comment by vikram-redhat** [comment]
@ahardin-rh PTAL.

@xingxingxia or @xiaocwan - can you please provide QE review?

**Comment by xingxingxia** [comment]
> Using OC
> Using Kubectl

LGTM, only except the letters case for them similar to https://github.com/openshift/openshift-docs/pull/10842#discussion_r205062071

**Comment by juanvallejo** [comment]
@xingxingxia thanks, fixed

**Pull request closed without merging**